### PR TITLE
SERENITY-2004 update yieldCurve types and add validator

### PR DESCRIPTION
**Pull request by indra-cloudwall**

### CODE REVIEW

File: src/python/serenity_types/pricing/derivatives/rates/yield_curve.py [hunk 1]
```diff
@@ -142,6 +142,16 @@ class YieldCurveDefinition(BaseModel):
...
 
+    rate_source_types: List[RateSourceType]
+    """
+    For reference purposes, the list of rate sources that were used to bootstrap this curve.
+    """
+
+    interpolation_method: InterpolationMethod
+    """
+    The specific interpolator type used to bootstrap this curve.
+    """
```
Comment (indra-cloudwall) [inline src/python/serenity_types/pricing/derivatives/rates/yield_curve.py:153]: @kdowney-cloudwallcapital Should we move this to YieldCurveDefinition instead?
And from the code, seems like rate_source_types will always be just one value i.e. not a list, if that's so I can add the fix

File: src/python/serenity_types/pricing/derivatives/rates/yield_curve.py [hunk 2]
```diff
@@ -174,6 +184,12 @@ class RawYieldCurve(BaseModel):
     A term structure of yield curve inputs. The RAW representation is offered to clients so they
     can either do their own interpolation or for diagnostics.
     """
+
+    rate_source_types: List[RateSourceType]
```
Comment (indra-cloudwall) [inline src/python/serenity_types/pricing/derivatives/rates/yield_curve.py:188]: From the code, seems like RawYieldCurve will be generated based on rate_source_types.
And same comment with above

File: src/python/serenity_types/pricing/derivatives/rates/yield_curve.py (hunk 1 above)
Comment (kdowney-cloudwallcapital) [reply src/python/serenity_types/pricing/derivatives/rates/yield_curve.py:153]: No, in the most general case it is a list; we may piece together the curve from more than one source in the future.

Comment (kdowney-cloudwallcapital) [reply src/python/serenity_types/pricing/derivatives/rates/yield_curve.py:153]: And I am fine with this change to shift the metadata entirely to YieldCurveDefinition, it makes sense to me.

**Review by kdowney-cloudwallcapital — APPROVED** [review]

**Pull request merged**